e2e/node/basic/identity.test.ts

```diff
@@ -15,6 +20,12 @@ function createIdentity(seed: number): SignIdentity {
   return Ed25519KeyIdentity.generate(new Uint8Array(seed1));
 }
 
+function createSecp256k1Identity(seed?: number): SignIdentity {
+  const seed1 = new Array(32).fill(0);
```

Comment:
Making the case for using a seed - wouldn't it allow someone to use an additional tool to turn a seed phrase into an IC-compatible secp256k1Identity? I think that's a use case that groups like Plug would be interested in

Reply:
Also copying my comment below here and deleting it (to have the full context in this thread): 

> Should we omit the optional seed argument, or is it needed somewhere? 
> * It's a bit strange that even though the seed is passed in, the private key could still use some other random bytes (line 64). 
> * It seems safer to me from a security perspective if the seed cannot be provided at all, and the method guarantees that the key generation happens appropriately. WDYT? 

Regarding your comment above: 
* The problem with the current implementation is that it is not guaranteed that the key is actually derived from the seed. The reason is that if `!privateKeyVerify`, the key is still freshly random, and thus for the same seed different keys could be generated. 
* I suggested the simplest solution, which is just not supporting generation from a seed. ;-) But if it is a requirement to support this, see the following: 
* From [the code](https://github.com/cryptocoinjs/secp256k1-node/blob/024fbdad3fb64499e8db8b35246c2f0a36afa8c8/lib/elliptic.js#L76) and [tests](https://github.com/cryptocoinjs/secp256k1-node/blob/024fbdad3fb64499e8db8b35246c2f0a36afa8c8/test/privatekey.js#L4) it seems that the key must be 32 bytes and `n` and `zero` are not allowed. 
* So what we could do is to throw an error in this method in case a seed is given and `!privateKeyVerify`. But this would definitely require a good documentation comment of this method, to make it clear to the caller that some seeds are not accepted. 

WDYT? 

@Dfinity-Bjoern: do you have an opinion on this?

Reply:
Based on this feedback, I've removed the secret key and added a test for the `fromSecretKey` method. I'll also add tests to ensure safety in the `fromSecretKey` input content

Reply:
Thanks for adapting this!